lib/analyse.ml

```diff
@@ -236,8 +274,11 @@ module Analysis = struct
     else if List.filter is_toplevel opam_files = [] then Lwt_result.fail (`Msg "No top-level opam files found!")
     else (
       begin
-        if is_duniverse then duniverse_selections ~job ~platforms dir
-        else opam_selections ~solver ~job ~platforms ~opam_repository_commit ~opam_files dir
+        match ty with
+        | `Duniverse -> duniverse_selections ~job ~platforms dir
+        | `Ocaml_repo -> opam_selections ~solver ~job ~platforms ~opam_repository_commit ~opam_files dir
+        | `Ocaml_compiler -> ocaml_compiler_selections ~platforms
+        | `Opam | `Dune -> failwith "TODO"
```

Comment:
Going to test out the OCaml compiler support before adding in runes for Opam and Dune (both of which have special bootstrap requirements too)